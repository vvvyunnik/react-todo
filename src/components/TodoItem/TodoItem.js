import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import { updateTodo, removeTodo } from "../../store/todoSlice";
import { ReactComponent as CrossIcon } from "../../assets/cross.svg";
import Button from "../shared/Button/Button";
import "./TodoItem.scss";

function TodoItem({ id, text, completed }) {
  const dispatch = useDispatch();

  const handleDeleteTodo = () => {
    dispatch(removeTodo(id));
  };

  const handleToggleComplete = () => {
    dispatch(updateTodo({ id, text, completed: !completed }));
  };

  const itemClass = `todo-item ${completed ? "completed" : ""}`;

  return (
    <li className={itemClass}>
      <input
        type="checkbox"
        className="todo-item-checkbox"
        checked={completed}
        onChange={handleToggleComplete}
      />
      <span className="todo-item-text">{text}</span>
      <Button className="delete-btn" onClick={handleDeleteTodo}>
        <CrossIcon />
      </Button>
    </li>
  );
}

TodoItem.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
};

export default TodoItem;
