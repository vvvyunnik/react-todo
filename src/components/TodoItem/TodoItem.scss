@use '../../styles/variables.scss' as *;

.todo-item {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid $light-color;
    margin-top: 8px;
    padding: 2px;

    &.completed {
        opacity: 0.5;
        text-decoration: line-through;
    }

    .todo-item-checkbox {
        width: 25px;
        height: 25px;
        background-color: #ffffff;
        border-radius: 50%;
        border: 1px solid $text-secondary-color;
        appearance: none;
        outline: none;
        cursor: pointer;

        &:checked {
            background-color: $text-secondary-color;
        }
    }

    .todo-item-text {
        color: $text-primary-color
    }
}