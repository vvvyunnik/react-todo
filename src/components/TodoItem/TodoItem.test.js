import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import TodoItem from "./TodoItem";

jest.mock("../../store/todoSlice.js", () => ({
  updateTodo: jest.fn(),
  removeTodo: jest.fn(),
}));

const mockStore = configureStore([]);

const renderTodoItemWithState = (initialState) => {
  const store = mockStore(initialState);
  return render(
    <Provider store={store}>
      <TodoItem id="1" text="Sample Todo" completed={false} />
    </Provider>
  );
};

describe("TodoItem", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test("matches snapshot", () => {
    const initialState = {
        todo: {
          items: [{ id: "1", text: "Sample Todo", completed: false }],
        },
      };

    const todoItemElement = renderTodoItemWithState(initialState);
    expect(todoItemElement).toMatchSnapshot();
  });
});
