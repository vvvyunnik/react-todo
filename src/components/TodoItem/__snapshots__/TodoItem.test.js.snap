// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="todo-item "
      >
        <input
          class="todo-item-checkbox"
          type="checkbox"
        />
        <span
          class="todo-item-text"
        >
          Sample Todo
        </span>
        <button
          class="no-style-btn delete-btn"
          type="button"
        >
          <svg>
            cross.svg
          </svg>
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="todo-item "
    >
      <input
        class="todo-item-checkbox"
        type="checkbox"
      />
      <span
        class="todo-item-text"
      >
        Sample Todo
      </span>
      <button
        class="no-style-btn delete-btn"
        type="button"
      >
        <svg>
          cross.svg
        </svg>
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
